sdp/processors/datasets/librispeech/create_initial_manifest.py

```diff
@@ -0,0 +1,137 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import fnmatch
+import glob
+import json
+import os
+import typing as tp
+
+from sdp.processors.base_processor import BaseProcessor
+from sdp.utils.common import download_file, extract_archive
+
+
+def get_librispeech_url_list(names: list[str]) -> list[str]:
+    urls = [
+        "https://openslr.org/resources/12/dev-clean.tar.gz",
+        "https://openslr.org/resources/12/dev-other.tar.gz",
+        "https://openslr.org/resources/12/test-clean.tar.gz",
+        "https://openslr.org/resources/12/test-other.tar.gz",
+        "https://openslr.org/resources/12/train-clean-100.tar.gz",
+        "https://openslr.org/resources/12/train-clean-360.tar.gz",
+        "https://openslr.org/resources/12/train-other-500.tar.gz",
+    ]
+    if "all" not in names:
+        filtered_urls = [url for url in urls if url.split('/')[-1].split('.tar')[0] in names]
+    else:
+        filtered_urls = urls
+
+    if len(filtered_urls) == 0:
+        raise ValueError("No data found")
+
+    return filtered_urls
+
+
+class CreateInitialManifestLibrispeech(BaseProcessor):
+    """Processor to create initial manifest for the Librispeech dataset.
+
+    Dataset link: https://openslr.org/12
+
+    Will download all files, extract tars and create manifest file with the
+    'audio_filepath' and 'text' fields
+
+    Args:
+        names (list[str]): Which data sets or their combinations shoudld be processed
+            - options are:
+            ["dev-clean"],
+            ["dev-other"],
+            ["test-clean"],
+            ["test-other"],
+            ["train-clean-100"],
+            ["train-clean-360"],
+            ["train-other-500"],
+            ["all"] (for all datasets avalable)
+
+        raw_data_dir (str): Path to folder where should the filed be donwloaded and extracted
+
+    Returns:
+       This processor generates an initial manifest file with the following fields::
+
+            {
+                "audio_filepath": <path to the audio file>,
+                "text": <transcription>,
+            }
+    """
+
+    def __init__(
+        self,
+        names: list[str],
+        raw_data_dir: str,
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.names = names
+        self.raw_data_dir = raw_data_dir
+
+    def process_transcrip(self, file_path: str) -> list[dict[str, tp.Any]]:
+        """Parse transcript file and put it inside manyfest
+        We assume that flac files are located in the same directory as transcript file.
+        """
+
+        entries = []
+        root = os.path.dirname(file_path)
+
+        with open(file_path, encoding="utf-8") as fin:
+            for line in fin:
+                id, text = line[: line.index(" ")], line[line.index(" ") + 1 :]
+                transcript_text = text.lower().strip()
```

Comment:
Does librispeech contain cased letters? If yes, then best not to lose this information. So I would not call `.lower()` here - the user can use `SubMakeLowercase` later if they want this to be lowercased

Reply:
Librispeech transcribe is in uppercases. E.g. 
"text": "THIS BOND IS BETWEEN MAN AND MAN IS USUALLY FORMED IN EARLY YOUTH AND CAN ONLY BE BROKEN BY DEATH"
That is why I used .lower() here. Should I keep it or remove it?

Reply:
removed it